models/desired_lrp_test.go

```diff
@@ -1471,6 +1497,85 @@ var _ = Describe("DesiredLRPUpdate", func() {
 			})
 		})
 	})
+
+	Describe("IsImageCredentialsUpdated", func() {
+		var existingLRP *models.DesiredLRP
+
+		BeforeEach(func() {
+			existingLRP = model_helpers.NewValidDesiredLRP("some-guid")
+			existingLRP.ImageUsername = "existing-username"
+			existingLRP.ImagePassword = "existing-password"
+		})
+
+		Context("when the update does not contain image credentials", func() {
+			It("returns false", func() {
+				update := &models.DesiredLRPUpdate{}
+				Expect(update.IsImageCredentialsUpdated(existingLRP)).To(BeFalse())
+			})
+		})
+
+		Context("when only username is provided", func() {
+			It("returns false", func() {
+				update := &models.DesiredLRPUpdate{}
+				update.SetImageUsername("new-username")
+				Expect(update.IsImageCredentialsUpdated(existingLRP)).To(BeFalse())
+			})
+		})
+
+		Context("when only password is provided", func() {
+			It("returns false", func() {
+				update := &models.DesiredLRPUpdate{}
+				update.SetImagePassword("new-password")
+				Expect(update.IsImageCredentialsUpdated(existingLRP)).To(BeFalse())
+			})
+		})
+
+		Context("when both username and password are provided and differ", func() {
+			It("returns true when username differs", func() {
+				update := &models.DesiredLRPUpdate{}
+				update.SetImageUsername("new-username")
+				update.SetImagePassword("existing-password")
+				Expect(update.IsImageCredentialsUpdated(existingLRP)).To(BeTrue())
+			})
+
+			It("returns true when password differs", func() {
+				update := &models.DesiredLRPUpdate{}
+				update.SetImageUsername("existing-username")
+				update.SetImagePassword("new-password")
+				Expect(update.IsImageCredentialsUpdated(existingLRP)).To(BeTrue())
+			})
+
+			It("returns true when both differ", func() {
+				update := &models.DesiredLRPUpdate{}
+				update.SetImageUsername("new-username")
+				update.SetImagePassword("new-password")
+				Expect(update.IsImageCredentialsUpdated(existingLRP)).To(BeTrue())
+			})
+		})
+
+		Context("when both username and password are provided and are equal", func() {
+			It("returns false", func() {
+				update := &models.DesiredLRPUpdate{}
+				update.SetImageUsername("existing-username")
+				update.SetImagePassword("existing-password")
+				Expect(update.IsImageCredentialsUpdated(existingLRP)).To(BeFalse())
+			})
+		})
+
+		Context("when existing LRP has no credentials", func() {
+			BeforeEach(func() {
+				existingLRP.ImageUsername = ""
+				existingLRP.ImagePassword = ""
+			})
+
+			It("returns true when new credentials are provided", func() {
```

Comment:
Can you add tests here for no existing credentials + no updated creds, only username, and only password returning false?